### PR TITLE
TwelveDaysTest Update

### LINKED ISSUE #1135: twelve-days: update tests and add version file
The twelve-days tests should be updated to exactly match the canonical data. Also a version file should be added to match the canonical data version.

### PR DESCRIPTION
**Pull request by Kyle-P**
Revised TwelveDaysTest according to canonical data and added version file.

Closes #1135 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java [hunk 1]
```diff
@@ -22,16 +22,16 @@ public void testVerseOne() {
     @Ignore("Remove to run test")
     @Test
     public void testVerseTwo() {
-        String expectedVerseTwo = "On the second day of Christmas my true love gave to me, two Turtle Doves, " +
+        String expectedVerseTwo = "On the second day of Christmas my true love gave to me, " + "two Turtle Doves, " +
```
Comment (FridaTveit) [inline exercises/twelve-days/src/test/java/TwelveDaysTest.java]: These changes aren't needed. It doesn't matter where you split up the string. Normally you would only do it if it doesn't all fit on one line. Does that make sense? 🙂

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java [hunk 2]
```diff
@@ -119,14 +119,26 @@ public void testVerseTwelve() {
...
                 "On the third day of Christmas my true love gave to me, three French Hens, two Turtle Doves, " +
                 "and a Partridge in a Pear Tree.\n";
-        assertEquals(expectedVerseOneToThree, twelveDays.verses(1, 3));
+        assertEquals(expectedVersesOneToThree, twelveDays.verses(1, 3));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testFourthToSixthVerses() {
+        String expectedVersesFourToSix - "On the fourth day of Chrsitmas my true love gave to me, " +
```
Comment (FridaTveit) [inline exercises/twelve-days/src/test/java/TwelveDaysTest.java]: This should have `=` not `-`, that's why the build is failing 🙂

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java [hunk 3]
```diff
@@ -22,16 +22,16 @@ public void testVerseOne() {
...
     }
 
     @Ignore("Remove to run test")
     @Test
     public void testVerseThree() {
-        String expectedVerseThree = "On the third day of Christmas my true love gave to me, three French Hens, " +
-                "two Turtle Doves, and a Partridge in a Pear Tree.\n";
+        String expectedVerseThree = "On the third day of Christmas my true love gave to me, " + "three French Hens, " +
+                "two Turtle Doves, " + "and a Partridge in a Pear Tree.\n";
         assertEquals(expectedVerseThree, twelveDays.verse(3));
```
Comment (FridaTveit) [inline exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Same goes for this change 🙂

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java (hunk 1 above)
Comment (Kyle-P) [reply exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Yup. I'll fix that up. Just to clarify: split up strings only when it doesn't fit on one line or if it's over 120 characters?

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java (hunk 2 above)
Comment (Kyle-P) [reply exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Oops. I'll fix that up. I'll be sure to check my edits next time :)

**Comment by Kyle-P** [comment]
Thanks for reviewing @FridaTveit! I'll add a commit fixing everything up.

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java (hunk 1 above)
Comment (sjwarner-bp) [reply exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Splitting lines of code doesn't have a 'hard' rule, some prefer 120 characters, some prefer 80. Other people might even have more rules! [Here](https://stackoverflow.com/questions/3137778/how-and-where-to-break-long-lines-of-code) is a link that might go into some more info. As for here, I don't think there is a written rule around this. As long as it is sensible, it should be ok 🙂

Comment (Kyle-P) [reply exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Alright, thanks @sjwarner-bp!

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java [hunk 4]
```diff
@@ -119,14 +119,26 @@ public void testVerseTwelve() {
...
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testFourthToSixthVerses() {
+        String expectedVersesFourToSix = "On the fourth day of Chrsitmas my true love gave to me, " + 
+                "four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n\n" +
+                "On the fifth day of Christmas my true love gave to me, five Gold Rings, four Calling Birds, " + 
+                "three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n\n" +
+                "On the sixth day of Christmas my true love gave to me, six Geese-a-Laying, five Gold Rings, " + 
+                "four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n"
```
Comment (c-thornton) [inline exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Did a quick look over this. @Kyle-P you are missing a ; at the end of line 140. Hide and seek champion of the world. :)

Comment (Kyle-P) [reply exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Thanks @c-thornton! Nice catch!

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java [hunk 5]
```diff
@@ -119,14 +119,26 @@ public void testVerseTwelve() {
...
                 "On the third day of Christmas my true love gave to me, three French Hens, two Turtle Doves, " +
                 "and a Partridge in a Pear Tree.\n";
-        assertEquals(expectedVerseOneToThree, twelveDays.verses(1, 3));
+        assertEquals(expectedVersesOneToThree, twelveDays.verses(1, 3));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testFourthToSixthVerses() {
+        String expectedVersesFourToSix = "On the fourth day of Chrsitmas my true love gave to me, " +
```
Comment (c-thornton) [inline exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Nearly had it, missed a typo here on this line. Need to flip the s and i in Christmas and then it should be good.

Comment (Kyle-P) [reply exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Alright, thanks @c-thornton!

Comment (c-thornton) [reply exercises/twelve-days/src/test/java/TwelveDaysTest.java]: No problem, should hopefully pass all checks this time.

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java [hunk 6]
```diff
@@ -31,7 +31,7 @@ public void testVerseTwo() {
     @Test
     public void testVerseThree() {
         String expectedVerseThree = "On the third day of Christmas my true love gave to me, three French Hens, " +
-                "two Turtle Doves, and a Partridge in a Pear Tree.\n";
+                "two Turtle Doves, " + "and a Partridge in a Pear Tree.\n";
```
Comment (sjwarner-bp) [inline exercises/twelve-days/src/test/java/TwelveDaysTest.java]: I think you missed one of the `+` @FridaTveit mentioned :slightly_smiling_face:

**Comment by Kyle-P** [comment]
Oops. Thanks for catching that @sjwarner-bp!

File: exercises/twelve-days/src/test/java/TwelveDaysTest.java (hunk 6 above)
Comment (Kyle-P) [reply exercises/twelve-days/src/test/java/TwelveDaysTest.java]: Oops. Thanks for catching that!

**Pull request merged**